src/fish.cpp

```diff
@@ -512,6 +512,16 @@ int main(int argc, char **argv) {
     if (!opts.no_exec && !opts.no_config) {
         read_init(parser, paths);
     }
+
+    if (opts.is_interactive_session && opts.no_config && !opts.no_exec) {
+        // If we have no config, we default to the default key bindings.
+        parser.vars().set_one(L"fish_key_bindings", ENV_UNEXPORT, L"fish_default_key_bindings");
+        if (function_exists(L"fish_default_key_bindings", parser)) {
+            std::vector<std::string> cmd {"fish_default_key_bindings"};
```

Comment:
> The main remaining issue is that the colorscheme is entirely empty, so everything will be colored normal. We probably want to load the default theme, but I'm not sold on calling `fish_config theme choose fish\ default` here.
> We might want to hard-code the default theme in highlight.cpp directly

Hmm can we maybe reuse `share/functions/__fish_config_interactive.fish`?
We can call something like `__fish_config_interactive.fish --no-config`, which shall behave as normal, except that it skips the creation of config directories and config.fish at the very bottom?

AFAICT the main point of --no-config is to make scripts portable; interactive config is probably okay here.

Reply:
> (We can merge as is of course but we should clean up soon)

Is that cleanup straighforward? I recommend pushing to the next version if not.

Reply:
>Is that cleanup straighforward? I recommend pushing to the next version if not.

Push it. Push it real good.